src/main/java/Problem1/BinarySearch.java

```diff
@@ -1,11 +1,23 @@
 package Problem1;
 
 public class BinarySearch {
-    // This is to be done recursively
-    // Do not change signature (return type, function name, parameter types)
-    // Add a helper function for recursion calls
     public static int binarySearch(int[] data, int target) {
-        // homework
-        return -1;  // place holder
+        return binarySearch(data, target, 0, data.length - 1);
+    }
+    private static int binarySearch(int[] data, int target, int left, int right) {
+        int middle = left + (right - left) / 2;
+        if (right < left) {
+            return -1;
+        }
+        if (target < data[middle]) {
+            return binarySearch(data, target, left, middle - 1);
+        }
+        if (target > data[middle]) {
+            return binarySearch(data, target, middle +1, right);
+        }
+        if (target == data[middle]) {
```

Comment:
this check is not needed because there's only one possibility left after the checks on line 12 and 15.